webapp/src/main/java/com/box/l10n/mojito/service/branch/notification/slack/BranchNotificationMessageSenderSlack.java

```diff
@@ -126,6 +131,17 @@ public void sendScreenshotMissingMessage(String branchName, String messageId, St
     }
   }
 
+  public boolean isUsernameBlocked(String username) {
+    for (String blockedUsernameRegex : blockedUsernames) {
+      if (Pattern.compile(blockedUsernameRegex, Pattern.CASE_INSENSITIVE)
```

Comment:
We should do the `Pattern.compile` once and then cache the matcher, rather than compile the pattern on each call to `isUsernameBlocked`

Reply:
Done, thanks!